iredis/entry.py

```diff
@@ -233,9 +248,172 @@ def gather_args(ctx, h, p, n, password, newbie, iredisrc, decode, raw, rainbow,
     if rainbow is not None:
         config.rainbow = rainbow
 
+    # gather connection_kwargs
+    config.connection_class = Connection
+    config.connection_kwargs = {}
+    config.connection_kwargs["host"] = h or config.hostname
+    config.connection_kwargs["port"] = p or config.port
+    config.connection_kwargs["password"] = password or config.password
+    config.connection_kwargs["socket_keepalive"] = config.socket_keepalive
+
+    if config.decode:
+        config.connection_kwargs["encoding"] = config.decode
+        config.connection_kwargs["decode_responses"] = True
+        config.connection_kwargs["encoding_errors"] = "replace"
+    else:
+        config.connection_kwargs["decode_responses"] = False
+
+    dsn_uri = None
+    if config_obj["alias_dsn"] and dsn:
+        try:
+            dsn_uri = config_obj["alias_dsn"].get(dsn)
+        except KeyError:
+            click.secho(
+                "Could not find the specified DSN in the config file. "
+                'Please check the "[alias_dsn]" section in your '
+                "iredisrc.",
+                err=True,
+                fg="red",
+            )
+            exit(1)
+
+    if dsn_uri:
+        parse_url(dsn_uri)
+
     return ctx
 
 
+FALSE_STRINGS = ("0", "F", "FALSE", "N", "NO")
+
+
+def to_bool(value):
+    if value is None or value == "":
+        return None
+    if isinstance(value, str) and value.upper() in FALSE_STRINGS:
+        return False
+    return bool(value)
+
+
+URL_QUERY_ARGUMENT_PARSERS = {
+    "socket_timeout": float,
+    "socket_connect_timeout": float,
+    "socket_keepalive": to_bool,
+    "retry_on_timeout": to_bool,
+    "max_connections": int,
+    "health_check_interval": int,
+    "ssl_check_hostname": to_bool,
+}
```

Comment:
Still, those options and `parse_url` is kind of heavy for iredis, I think we only need `host` `port` `path` `db` `scheme`. The others can be spicify by commendline.